code/server/src/FIFOHandler.cpp

```diff
@@ -0,0 +1,70 @@
+/*
+ * author: Elisa Maiettini
+ * date: 30/11/2016
+ */
+#include <FIFOHandler.hpp>
+
+ using namespace std;
+ using namespace yarp::os;
+
+ void FIFOHandler::initThread(){
+   cout << "FIFOHandler: thread inizialization" << endl;
+   FIFO = std::vector<int>();
+   fifo_mutex = yarp::os::Semaphore();
+ }
+
+ void FIFOHandler::run(){
+   fifo_mutex.wait();
+
+   cout << "FIFOHandler: the current FIFO is: {";
+   std::vector<int>::iterator it;
+   for(it = FIFO.begin(); it != FIFO.end() ; it++ ){
+      cout << *it << " ";
+   }
+   cout << "}" << endl;
+   fifo_mutex.post();
+ }
+
+ int FIFOHandler::getHead(){
+   fifo_mutex.wait();
+   int head = FIFO.front();
+   fifo_mutex.post();
+   return head;
+ }
+
+ vector<int> FIFOHandler::getFIFO(){
```

Comment:
actually I don't use it! It is a function that I used in a previous version of the project! I can delete it